### PR TITLE
Optimize noteram speed, copy all data at once

### PR DESCRIPTION
**Pull request by Gary-Hobson**
## Summary
sched_note_add copies one byte at a time
Refer to circbuff to copy the entire payload at once

### CODE REVIEW

**Comment by acassis** [comment]
@masayuki2009 I added you as reviewer because I didn't find Yuuichi Nakamura's github user

File: drivers/note/noteram_driver.c [hunk 1]
```diff
@@ -195,43 +195,6 @@ static void noteram_record_taskname(pid_t pid, const char *name)
...
- * Input Parameters:
- *   PID - Task ID
- *
- * Returned Value:
- *   None
- *
- ****************************************************************************/
-
-#if CONFIG_DRIVER_NOTERAM_TASKNAME_BUFSIZE > 0
-static void noteram_remove_taskname(pid_t pid)
```
Comment (xiaoxiang781216) [inline drivers/note/noteram_driver.c:213]: let's move the task name change to the new patch

**Comment by masayuki2009** [comment]
@YuuichiNakamura
Do you have any comments on this PR?

**Comment by Gary-Hobson** [comment]
This commit needs to be revised and will be resubmitted later

**Pull request closed without merging**